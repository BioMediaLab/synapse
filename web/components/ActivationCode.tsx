import React, { Component } from "react";
import {
  withStyles,
  createStyles,
  Theme,
  TextField,
  Button,
  FormControl,
  Select,
  InputLabel,
  MenuItem,
} from "@material-ui/core";
import SnackbarMessage from "./SnackbarMessage";
import gql from "graphql-tag";
import { Query, Mutation } from "react-apollo";

const GET_COURSES = gql`
  query {
    myCourseRoles {
      id
      course {
        id
        name
        requireActivation
      }
    }
    me {
      id
      name
    }
  }
`;

const ACTIVATION_CODE_USE_MUTATION = gql`
  mutation($activation_code: String!, $course_id: String!, $user_id: String!) {
    useActivationCode(
      activation_code: $activation_code
      course_id: $course_id
      user_id: $user_id
    ) {
      id
    }
  }
`;

const ACTIVATION_CODE_CLEAR_MUTATION = gql`
  mutation($activation_code: String!, $user_id: String!) {
    clearActivationCode(activation_code: $activation_code, user_id: $user_id) {
      id
    }
  }
`;

const ACTIVATION_CODE_CREATE_MUTATION = gql`
  mutation($activation_code: String!) {
    addActivationCode(activation_code: $activation_code) {
      id
    }
  }
`;

const styles = (theme: Theme) =>
  createStyles({
    button: {
      marginTop: theme.spacing.unit * 2,
      marginBottom: theme.spacing.unit,
      width: 80,
      alignSelf: "flex-end",
    },
    formControl: {
      marginTop: 0,
    },
    flexDiv: {
      display: "flex",
      flexDirection: "column",
      maxWidth: 400,
    },
  });

interface IActivationCodeProps {
  isAdmin: boolean;
  classes: {
    button: string;
    formControl: string;
    flexDiv: string;
  };
}

interface IActivationCodeState {
  activationCode: string;
  newActivationCode: string;
  courseId: string;
  activationCodeToClear: string;
}

class ActivationCode extends Component<
  IActivationCodeProps,
  IActivationCodeState
> {
  state = {
    activationCode: "",
    newActivationCode: "",
    courseId: "",
    activationCodeToClear: "",
  };

  handleChange = name => event => {
    switch (name) {
      case "activationCode":
        this.setState({ activationCode: event.target.value });
        break;
      case "newActivationCode":
        this.setState({ newActivationCode: event.target.value });
        break;
      case "courseId":
        this.setState({ courseId: event.target.value });
        break;
      case "activationCodeToClear":
        this.setState({ activationCodeToClear: event.target.value });
        break;
      default:
        break;
    }
  };

  render() {
    const classes = this.props.classes;
    return (
      <div className={classes.flexDiv}>
        <Query query={GET_COURSES}>
          {({ loading, data }) => {
            if (loading) {
              return <div />;
            }
            const courses = data.myCourseRoles.map(role => role.course);
            return (
              <>
                <FormControl className={classes.formControl} fullWidth>
                  <InputLabel>Course to Activate</InputLabel>
                  <Select
                    value={this.state.courseId}
                    onChange={this.handleChange("courseId")}
                  >
                    {courses.map(course => {
                      return course.requireActivation ? (
                        <MenuItem key={course.id} value={course.id}>
                          {course.name}
                        </MenuItem>
                      ) : null;
                    })}
                  </Select>
                </FormControl>
                <Mutation mutation={ACTIVATION_CODE_USE_MUTATION}>
                  {(doMutation, mutationResult) => {
                    return (
                      <>
                        <TextField
                          label="Activation Code"
                          placeholder="****-****-****-****"
                          value={this.state.activationCode}
                          fullWidth
                          onChange={this.handleChange("activationCode")}
                        />
                        <Button
                          variant="contained"
                          className={classes.button}
                          onClick={event => {
                            doMutation({
                              variables: {
                                activation_code: this.state.activationCode,
                                course_id: this.state.courseId,
                                user_id: data.me.id,
                              },
                            });
                            this.setState({
                              activationCode: "",
                              courseId: "",
                            });
                          }}
                        >
                          Activate
                        </Button>
                        {mutationResult.data ? (
                          <SnackbarMessage
                            success
                            message="Course Activation Successful!"
                          />
                        ) : null}
                        <SnackbarMessage error message={mutationResult.error} />
                      </>
                    );
                  }}
                </Mutation>
                <Mutation mutation={ACTIVATION_CODE_CLEAR_MUTATION}>
                  {(doMutation, mutationResult) => {
                    return (
                      <>
                        <TextField
                          label="Reset Activation Code"
                          placeholder="****-****-****-****"
                          fullWidth
                          onChange={this.handleChange("activationCodeToClear")}
                          value={this.state.activationCodeToClear}
                        />

                        <Button
                          variant="contained"
                          className={classes.button}
                          onClick={event => {
                            doMutation({
                              variables: {
                                activation_code: this.state
                                  .activationCodeToClear,
                                user_id: data.me.id,
                              },
                            });

                            this.setState({ activationCodeToClear: "" });
                          }}
                        >
                          Reset
                        </Button>
                        {mutationResult.data ? (
                          <SnackbarMessage
                            success
                            message="Activation Code Reset Successful!"
                          />
                        ) : null}
                        <SnackbarMessage error message={mutationResult.error} />
                      </>
                    );
                  }}
                </Mutation>
              </>
            );
          }}
        </Query>

        {this.props.isAdmin ? (
          <>
            <Mutation mutation={ACTIVATION_CODE_CREATE_MUTATION}>
              {(doMutation, mutationResult) => {
                return (
                  <>
                    <TextField
                      label="Create Activation Code"
                      placeholder="****-****-****-****"
                      fullWidth
                      onChange={this.handleChange("newActivationCode")}
                      value={this.state.newActivationCode}
                    />

                    <Button
                      variant="contained"
                      className={classes.button}
                      onClick={event => {
                        doMutation({
                          variables: {
                            activation_code: this.state.newActivationCode,
                          },
                        });

                        this.setState({ newActivationCode: "" });
                      }}
                    >
                      Create
                    </Button>
                    {mutationResult.data ? (
                      <SnackbarMessage
                        success
                        message="Activation Code Creation Successful!"
                      />
                    ) : null}
                    <SnackbarMessage error message={mutationResult.error} />
                  </>
                );
              }}
            </Mutation>
          </>
        ) : (
          <div />
        )}
      </div>
    );
  }
}

export default withStyles(styles)(ActivationCode);
