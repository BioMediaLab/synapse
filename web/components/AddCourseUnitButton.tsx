import React, { Component } from "react";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Switch from "@material-ui/core/Switch";
import gql from "graphql-tag";
import { Mutation } from "react-apollo";

const ADD_COURSE_UNIT = gql`
  mutation AddCourseUnit(
    $course_id: ID!
    $name: String!
    $description: String!
    $visible: Boolean
  ) {
    addCourseUnit(
      course_id: $course_id
      name: $name
      description: $description
      visible: $visible
    ) {
      id
      name
    }
  }
`;

class AddCourseUnitButton extends Component<{ courseId: string }> {
  state = {
    open: false,
    unit: {
      name: "",
      description: "",
      visible: false,
    },
  };

  handleClickOpen = () => {
    this.setState({ open: true });
  };

  handleClose = () => {
    this.setState({ open: false });
  };

  handleInputChange = name => event => {
    this.setState({
      unit: {
        ...this.state.unit,
        [name]: event.target.value,
      },
    });
  };

  handleToggleVisibility = () => {
    this.setState({
      unit: {
        ...this.state.unit,
        visibile: !this.state.unit.visible,
      },
    });
  };

  handleAddUnitToCourse = () => {
    console.warn("unimpl...");
  };

  render() {
    return (
      <Mutation mutation={ADD_COURSE_UNIT}>
        {addCourseUnit => (
          <div>
            <Button variant="contained" onClick={this.handleClickOpen}>
              Add Unit
            </Button>

            <Dialog
              open={this.state.open}
              onClose={this.handleClose}
              aria-labelledby="form-dialog-title"
            >
              <DialogTitle id="form-dialog-title">
                Add unit to course
              </DialogTitle>
              <DialogContent>
                <DialogContentText>
                  You can organize your course content by creating Units and
                  placing relevant course media inside. Fill out the settings
                  below and then hit the ADD button. Don't worry you can always
                  edit the settings later.
                </DialogContentText>

                <TextField
                  autoFocus
                  margin="dense"
                  id="name"
                  label="Name"
                  type="text"
                  fullWidth
                  onChange={this.handleInputChange("name")}
                />

                <TextField
                  autoFocus
                  margin="dense"
                  id="description"
                  label="Description"
                  type="text"
                  fullWidth
                  multiline
                  onChange={this.handleInputChange("description")}
                />

                <FormControlLabel
                  control={
                    <Switch
                      checked={this.state.unit.visible}
                      onChange={this.handleToggleVisibility}
                      value={this.state.unit.visible}
                      color="primary"
                    />
                  }
                  label="Visible"
                />
              </DialogContent>
              <DialogActions>
                <Button onClick={this.handleClose} color="primary">
                  Cancel
                </Button>
                <Button
                  onClick={() => {
                    addCourseUnit({
                      variables: {
                        course_id: this.props.courseId,
                        name: this.state.unit.name,
                        description: this.state.unit.description,
                        visibile: this.state.unit.visible,
                      },
                    });
                    this.handleClose();
                  }}
                  color="primary"
                >
                  Add
                </Button>
              </DialogActions>
            </Dialog>
          </div>
        )}
      </Mutation>
    );
  }
}

export default AddCourseUnitButton;
