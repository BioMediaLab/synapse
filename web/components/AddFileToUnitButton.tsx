import React from "react";
import gql from "graphql-tag";
import { Mutation } from "react-apollo";
import FileStackMaterial from "./FileStackMaterial";

const ADD_FILE_TO_UNIT = gql`
  mutation(
    $course_id: String!
    $unit_id: String!
    $name: String!
    $url: String!
    $type: String!
    $desc: String
  ) {
    createContent(
      name: $name
      type: $type
      url: $url
      course_id: $course_id
      unit_id: $unit_id
      description: $desc
    ) {
      id
    }
  }
`;

interface IFileUploadButtonProps {
  courseId: string;
  unitId: string;
}

class AddFileToUnitButton extends React.Component<IFileUploadButtonProps> {
  render() {
    console.log("unitId", this.props.unitId);
    return (
      <Mutation mutation={ADD_FILE_TO_UNIT}>
        {(doMutation, result) => {
          return (
            <FileStackMaterial
              onUploadComplete={result => {
                if (result.filesFailed && result.filesFailed.length > 0) {
                  console.warn(result.filesFailed);
                  throw new Error("file failed to upload");
                }

                result.filesUploaded
                  .filter(file => {
                    if (file.filename && file.url && file.mimetype) {
                      return true;
                    }
                    throw new Error("file property missing");
                  })
                  .forEach(async file => {
                    const uploadResult = await doMutation({
                      variables: {
                        name: file.filename,
                        course_id: this.props.courseId,
                        unit_id: this.props.unitId,
                        url: file.url,
                        type: file.mimetype,
                      },
                    });
                    if (uploadResult && uploadResult.errors) {
                      throw new Error(`Uploading ${file.filename} failed`);
                    }
                  });
              }}
              disabled={result.loading}
            />
          );
        }}
      </Mutation>
    );
  }
}

export default AddFileToUnitButton;
