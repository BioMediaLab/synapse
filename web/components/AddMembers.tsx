import React, { FunctionComponent, useState, ReactNode } from "react";
import {
  Button,
  Dialog,
  DialogTitle,
  DialogContent,
  DialogActions,
} from "@material-ui/core";
import { withSnackbar, InjectedNotistackProps, VariantType } from "notistack";

import { IUser } from "../queries/userQueries";
import UserSearch from "./UserSearch";

interface IMutationState {
  success: boolean;
  message?: string;
}

interface IProps {
  buttonText?: string;
  titleText?: string;
  children?: ReactNode;
  onComplete: (users: IUser[]) => Promise<IMutationState | null>;
}

type Props = IProps & InjectedNotistackProps;

const AddMembers: FunctionComponent<Props> = (props: Props) => {
  const [isOpen, setIsOpen] = useState(false);
  const [selectedUsers, updateUsers] = useState<IUser[]>([]);

  const buttonText = props.buttonText ? props.buttonText : "Add Members";
  const titleText = props.titleText ? props.titleText : "Add Members";
  const child = props.children ? props.children : <span />;

  return (
    <>
      <Button onClick={() => setIsOpen(true)}>{buttonText}</Button>
      <Dialog
        open={isOpen}
        onClose={() => setIsOpen(false)}
        fullWidth
        maxWidth="md"
      >
        <DialogTitle>{titleText}</DialogTitle>
        <DialogContent>
          <UserSearch onValueChange={users => updateUsers(users)} />
          {child}
        </DialogContent>
        <DialogActions>
          <Button
            onClick={async () => {
              setIsOpen(false);
              const users = selectedUsers;
              updateUsers([]);
              const complete = await props.onComplete(users);
              if (complete) {
                const opts = {
                  variant: (complete.success
                    ? "success"
                    : "fail") as VariantType,
                };
                if (complete.message) {
                  props.enqueueSnackbar(complete.message, opts);
                } else {
                  props.enqueueSnackbar(
                    complete.success ? "Success" : "Fail",
                    opts,
                  );
                }
              }
            }}
          >
            Confirm
          </Button>
          <Button
            onClick={() => {
              setIsOpen(false);
              updateUsers([]);
            }}
          >
            Cancel
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
};

export default withSnackbar(AddMembers);
