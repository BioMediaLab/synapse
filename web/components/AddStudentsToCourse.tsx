import React from "react";
import {
  withStyles,
  Button,
  Dialog,
  DialogTitle,
  DialogContent,
  DialogActions,
  Typography,
  Select,
  MenuItem,
  createStyles,
} from "@material-ui/core";
import { graphql, ChildMutateProps } from "react-apollo";
import gql from "graphql-tag";
import { withSnackbar, InjectedNotistackProps } from "notistack";

import UserSearch from "./UserSearch";
import { CourseRoleType, ADD_USERS_TO_COURSE } from "../queries/courseQueries";

const styles = theme =>
  createStyles({
    addDialog: {
      minWidth: theme.spacing.unit * 100,
      minHeight: theme.spacing.unit * 15,
    },
    fullDialog: {
      minHeight: theme.spacing.unit * 20,
    },
    roleSelect: {
      marginLeft: theme.spacing.unit,
      marginRight: theme.spacing.unit,
    },
  });

interface IStyles {
  classes: {
    addDialog: string;
    fullDialog: string;
    roleSelect: string;
  };
}

interface IProps {
  curCourseId: string;
  userAddCallback?: (userIds: string[]) => void;
}

interface IState {
  addDialogOpen: boolean;
  usersToAdd: any[];
  userType: CourseRoleType;
}

interface IVars {
  courseId: string;
  users: Array<{ user_id: string; role: string }>;
}

type Props = ChildMutateProps<
  IProps & IStyles & InjectedNotistackProps,
  { course: { id: string } },
  IVars
>;

class AddStudentsToCourse extends React.Component<Props, IState> {
  state = {
    addDialogOpen: false,
    usersToAdd: [],
    userType: "PROFESSOR" as CourseRoleType,
  };

  showAddDialog = () => {
    this.setState(state => ({ ...state, addDialogOpen: true }));
  };

  hideAddDialog = () => {
    this.setState(state => ({
      ...state,
      usersToAdd: [],
      addDialogOpen: false,
    }));
  };

  doUserAdd = async () => {
    this.hideAddDialog();
    if (this.props.curCourseId && this.state.usersToAdd.length > 0) {
      const res = await this.props.mutate({
        // TODO: support adding users of different types
        variables: {
          courseId: this.props.curCourseId,
          users: this.state.usersToAdd.map(user => ({
            user_id: user.id,
            role: this.state.userType,
          })),
        },
      });
      if (res && !res.errors) {
        this.props.enqueueSnackbar("Users added", { autoHideDuration: 1000 });
        if (this.props.userAddCallback) {
          this.props.userAddCallback(this.state.usersToAdd);
        }
        return;
      }
    }
    this.props.enqueueSnackbar("An Error Ocurred.", {
      autoHideDuration: 1000,
      variant: "error",
    });
  };

  updateSelectedUsers = users => {
    this.setState(state => ({ ...state, usersToAdd: users }));
  };

  updateSelectUserType = event => {
    this.setState(state => ({ ...state, userType: event.target.value }));
  };

  render() {
    return (
      <>
        <Dialog
          open={this.state.addDialogOpen}
          onClose={this.hideAddDialog}
          maxWidth="md"
          className={this.props.classes.fullDialog}
        >
          <DialogTitle>Choose new members</DialogTitle>
          <DialogContent className={this.props.classes.addDialog}>
            <UserSearch onValueChange={this.updateSelectedUsers} />
            <Typography
              variant="body1"
              style={{
                display: "flex",
                alignItems: "center",
              }}
            >
              <span>Add these users as</span>
              <Select
                className={this.props.classes.roleSelect}
                value={this.state.userType}
                onChange={this.updateSelectUserType}
              >
                <MenuItem value="PROFESSOR">professors</MenuItem>
                <MenuItem value="ADMIN">administrators</MenuItem>
                <MenuItem value="STUDENT">students</MenuItem>
                <MenuItem value="ASSISTANT">teaching assistants</MenuItem>
                <MenuItem value="AUDITOR">auditors</MenuItem>
              </Select>
              <span>into the class.</span>
            </Typography>
          </DialogContent>
          <DialogActions>
            <Button
              color="primary"
              onClick={this.doUserAdd}
              disabled={this.state.usersToAdd.length === 0}
            >
              Save
            </Button>
            <Button onClick={this.hideAddDialog}>Cancel</Button>
          </DialogActions>
        </Dialog>
        <Button color="primary" onClick={this.showAddDialog}>
          Add new members
        </Button>
      </>
    );
  }
}

export default graphql<IProps, {}, IVars>(ADD_USERS_TO_COURSE)(
  withSnackbar(withStyles(styles)(AddStudentsToCourse)),
);
