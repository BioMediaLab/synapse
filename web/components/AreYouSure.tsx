import React, { ReactNode } from "react";
import {
  DialogTitle,
  Dialog,
  DialogActions,
  Button,
  DialogContent,
} from "@material-ui/core";

interface IProps {
  showing: boolean;
  yes: () => void;
  no: () => void;
  optionYesText?: string;
  optionNoText?: string;
  title?: string;
  children?: ReactNode;
}

const AreYouSure: React.SFC<IProps> = props => {
  const child = props.children ? props.children : <span>Are you sure?</span>;
  const title = props.title ? props.title : "Confirm?";
  const yesText = props.optionYesText ? props.optionYesText : "Yes";
  const noText = props.optionYesText ? props.optionNoText : "No";
  return (
    <Dialog open={props.showing}>
      <DialogTitle>{title}</DialogTitle>
      <DialogContent>{child}</DialogContent>
      <DialogActions>
        <Button onClick={() => props.yes()}>{yesText}</Button>
        <Button onClick={() => props.no()}>{noText}</Button>
      </DialogActions>
    </Dialog>
  );
};

export default AreYouSure;
