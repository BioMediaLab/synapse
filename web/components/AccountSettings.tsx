import React from "react";
import { Switch, Typography, FormControlLabel } from "@material-ui/core";
import { Query, Mutation } from "react-apollo";
import gql from "graphql-tag";
import AreYouSure from "./AreYouSure";
import ErrorMessage from "./ErrorMessage";
import ActivationCode from "./ActivationCode";

const ALL_ABOUT_ME = gql`
  query {
    me {
      id
      name
      acceptsEmails
      isAdmin
    }
  }
`;

const MUTATE_ME = gql`
  mutation($acceptsEmails: Boolean, $bio: String) {
    updateUserSettings(fields: { acceptsEmails: $acceptsEmails, bio: $bio }) {
      id
      acceptsEmails
      bio
      name
    }
  }
`;

interface IState {
  sendEmailConfirm: boolean;
}

class AccountSettings extends React.Component<{}, IState> {
  state = {
    sendEmailConfirm: false,
  };

  render() {
    return (
      <>
        <Query query={ALL_ABOUT_ME}>
          {({ loading, error, data }) => {
            if (loading) {
              return <div>loading...</div>;
            }
            if (error) {
              return <ErrorMessage message={error.message} />;
            }
            return (
              <div>
                <Mutation mutation={MUTATE_ME}>
                  {(doMutation, { loading: muteLoad }) => {
                    return (
                      <>
                        <FormControlLabel
                          control={
                            <Switch
                              disabled={muteLoad}
                              checked={data.me.acceptsEmails}
                              onChange={event => {
                                const checked = event.target.checked;
                                if (!checked) {
                                  this.setState(state => ({
                                    ...state,
                                    sendEmailConfirm: true,
                                  }));
                                  return;
                                }
                                doMutation({
                                  variables: { acceptsEmails: true },
                                });
                              }}
                            />
                          }
                          label="Recieve emails from Synapse"
                        />
                        <AreYouSure
                          showing={this.state.sendEmailConfirm}
                          yes={() => {
                            this.setState(state => ({
                              ...state,
                              sendEmailConfirm: false,
                            }));
                            doMutation({ variables: { acceptsEmails: false } });
                          }}
                          no={() =>
                            this.setState(state => ({
                              ...state,
                              sendEmailConfirm: false,
                            }))
                          }
                        >
                          <Typography variant="body2">
                            Are you sure that you wish to stop seeing email
                            notifications from Synapse?
                          </Typography>
                        </AreYouSure>
                      </>
                    );
                  }}
                </Mutation>
                <ActivationCode />
              </div>
            );
          }}
        </Query>
      </>
    );
  }
}

export default AccountSettings;
