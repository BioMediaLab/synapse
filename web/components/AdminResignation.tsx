import React, { useState } from "react";
import { Mutation, Query } from "react-apollo";
import Button from "@material-ui/core/Button";

import { Router } from "../Router";
import { destroySessionFrontend } from "../lib/handleSessions";
import { REMOVE_ADMIN, MY_ID } from "../queries/userQueries";
import AreYouSure from "./AreYouSure";

const AdminResign = () => {
  const [showing, updateVisibility] = useState(false);

  return (
    <Query query={MY_ID}>
      {({ loading, error, data }) => {
        if (loading || error) {
          return <div />;
        }
        return (
          <Mutation mutation={REMOVE_ADMIN} variables={{ myId: data.me.id }}>
            {doResign => (
              <>
                <Button
                  variant="contained"
                  color="secondary"
                  onClick={() => updateVisibility(true)}
                >
                  Resign As Admin
                </Button>
                <AreYouSure
                  title="Permanently remove administrative access from your account?"
                  showing={showing}
                  yes={() => {
                    updateVisibility(false);
                    doResign();
                    Router.pushRoute("/");
                    destroySessionFrontend();
                  }}
                  no={() => updateVisibility(false)}
                >
                  You will be logged out. Click "yes" to proceed or "no" to
                  cancel.
                </AreYouSure>
              </>
            )}
          </Mutation>
        );
      }}
    </Query>
  );
};

export default AdminResign;
