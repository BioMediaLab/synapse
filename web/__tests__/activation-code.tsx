import "react-testing-library/cleanup-after-each";
import React from "react";
import { render } from "react-testing-library";
import ActivationCode from "../components/ActivationCode";
import { GET_COURSES } from "../components/ActivationCode";
import { MockedProvider } from "react-apollo/test-utils";

const mocks = [
  {
    request: {
      query: GET_COURSES,
    },
    result: {
      data: {
        myCourseRoles: {
          id: "testID01",
          course: {
            id: "testID02",
            name: "CourseName",
            requireActivation: true,
          },
        },
        me: {
          id: "testID03",
          name: "UserName",
        },
      },
    },
  },
];

test("renders", () => {
  const { container } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ActivationCode isAdmin />
    </MockedProvider>,
  );
  expect(container).toMatchSnapshot();
});
