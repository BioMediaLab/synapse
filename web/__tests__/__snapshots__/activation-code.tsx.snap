// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div
    class="ActivationCode-flexDiv-3"
  >
    <div />
    <div
      class="MuiFormControl-root-4 MuiFormControl-fullWidth-7"
    >
      <label
        class="MuiFormLabel-root-19 MuiInputLabel-root-8 MuiInputLabel-formControl-13 MuiInputLabel-animated-16"
        data-shrink="false"
      >
        Create Activation Code
      </label>
      <div
        class="MuiInputBase-root-39 MuiInput-root-26 MuiInput-underline-30 MuiInputBase-fullWidth-48 MuiInput-fullWidth-33 MuiInputBase-formControl-40 MuiInput-formControl-27"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input-49 MuiInput-input-34"
          placeholder="****-****-****-****"
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      class="MuiButtonBase-root-82 MuiButton-root-56 MuiButton-contained-67 MuiButton-raised-70 ActivationCode-button-1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label-57"
      >
        Create
      </span>
      <span
        class="MuiTouchRipple-root-91"
      />
    </button>
  </div>
</div>
`;
